packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -106,25 +107,34 @@ export function isTheia(): boolean {
     return false;
 }
 
-/*************************************************************************************************************
- * Get updated profile and session information
- * @param {sessNode} IZoweTreeNode
- *************************************************************************************************************/
-// This function does not perform any UI refresh; it just gets updated profile information.
-export async function refreshTree(sessNode: IZoweTreeNode) {
-    const profileType = sessNode.getProfile().type;
-    const allProf = Profiles.getInstance().getProfiles(profileType);
-    const baseProf = await Profiles.getInstance().getBaseProfile();
-    for (const profNode of allProf) {
-        if (sessNode.getProfileName() === profNode.name) {
-            setProfile(sessNode, profNode.profile);
-            const combinedSessionProfile = (await Profiles.getInstance().getCombinedProfile(profNode, baseProf))
-                .profile;
-            setSession(sessNode, combinedSessionProfile);
```

Comment:
The problem was here: `setSession(sessNode, combinedSessionProfile)`, where we manually updated session values, ignoring provided API to build session properly.